caom2/caom2/obs_reader_writer.py

```diff
@@ -244,6 +244,27 @@ def _get_child_text_as_boolean(self, element_tag, parent, ns, required):
         else:
             return child_element.text.lower() == "true"
 
+
+    def _add_keywords(self, keywords_list, element, ns, required):
+        """
+        Parses keywords sub-elements and adds them to a list
+
+        :param keywords_list: list to add the keywords to
+        :param element: xml element
+        :param ns: name space
+        :param required: keywords sub-element required or not
+        """
+        if self.version < 23:
+            keywords = self._get_child_text("keywords", element, ns, required)
+            if keywords is not None:
+                for keyword in keywords.split():
+                    keywords_list.add(keyword)
+        else:
+            keywords_element = self._get_child_element("keywords", element, ns, True)
```

Comment:
In the xsd most if not all caom2 entities with keywords use minOccurs=0, but required is set to True. Are empty elements always written then?